pilot/pkg/proxy/envoy/discovery.go

```diff
@@ -429,8 +429,9 @@ func debouncePush(startDebounce time.Time) {
 			since, time.Since(lastClearCache))
 		clearCacheMutex.Lock()
 		clearCacheTimerSet = false
-		lastClearCache = time.Now()
 		clearCacheMutex.Unlock()
+		// no need to protect, no data race in case DebounceAfter > 0
+		lastClearCache = time.Now()
```

Comment:
a time call under a lock is expensive indeed.  I am curious if the assertion made on line 433 is correct and if you could provide more details on why?

Cheers
-steve

Reply:
Certainly, `debouncePush` is called by `clearCache` and itself. 

When it is called by `clearCache` ,  `clearCacheTimerSet` is true, which means `clearCache` can not call `debouncePush` again.  And in this case `lastClearCache` is only read/write by `debouncePush`